SharpBucket/ISharpBucketRequester.cs

```diff
@@ -0,0 +1,21 @@
+﻿namespace SharpBucket
+{
+    /// <summary>
+    /// Interface that expose the methods that an end point should use to perform calls to the BitBucket API.
+    /// </summary>
+    public interface ISharpBucketRequester
+    {
+        string Get(string relativeUrl, object requestParameters = null);
+
+        T Get<T>(string relativeUrl, object requestParameters = null)
+            where T : new();
+
+        T Post<T>(T body, string relativeUrl)
```

Comment:
Should we expose a requestParameters parameter to Post Put and Delete methods before publicly exposing them through that interface ?

We have not need that until now, but if we need it one day it would cause a breaking change to that interface.
So maybe it's preferable to expose that even if we don't use it right now ?

Reply:
Another idea would be to expose in the interface the Send methods (which are currently private) instead of the Get,Post,Put,Delete methods
And to write the Get,Post,Put,Delete methods as extension methods of the interface

That pattern could be less subject to breaking changes, since if we need to add new overloads for the Post, Put or Delete, or even need to add support for another verb (like Patch which is not currently used by BitBucket but who know if they add it later)

Reply:
In fact there is a major issue with exposing the Send methods.
That methods needs a RestSharp.Method argument, and it would not be wise to publicly depends on RestSharp.

One solution would be to replace that parameter with the System.Net.Http.HttpMethod type in the interface, to avoid to publicly depends on RestSharp, and use only types provided by the framework.
But in that case, It will give impression that implementation should be done using HttpClient type, which may not be really better than depending on RestSharp type.

An other solution would be to replace that parameter with a simple string type like in System.Net.HttpWebRequest type on which RestSharp is based, but it can quickly lead to multiple allocation of the same strings that represent the verbs which will force us in definitive to use System.Net.Http.HttpMethod to pass the strings to the interface to always use the same string instances for the verbs (but without the help of the compiler to do that correctly).

And in both cases, as long as the implementation of that Send methods is done with RestSharp, we'll have to had a mapping to the ResSharp enum.

So what should we choose ?
 * Keep explicit Get,Post,Put,Delete methods in the interface and take the risk of a breaking change
 * Or go for exposing only Send methods in the interface but with which solution for the method parameter ?

I think I'll try with the reference to System.Net.Http.HttpMethod.